datafusion/functions/src/unicode/substr.rs

```diff
@@ -77,7 +78,11 @@ impl ScalarUDFImpl for SubstrFunc {
     }
 
     fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
-        utf8_to_str_type(&arg_types[0], "substr")
+        if arg_types[0] == DataType::Utf8View {
+            Ok(DataType::Utf8View)
+        } else {
+            utf8_to_str_type(&arg_types[0], "substr")
+        }
```

Comment:
This might be a temporary implementation in my opinion. We might need to modify `utf8_to_str_type` eventually.

Reply:
I agree

Reply:
Maybe we can make a second function like `optmized_utf8_to_str_type` -- with comments that make it clear that the difference between that and `utf8_to_str_type` if if the string function has a specialized implementation for string view.

Reply:
I added `optmized_utf8_to_str_type` in utils.rs, wonder if it's proper.